### PR TITLE
Update Test Vectors for RFC

### PR DESCRIPTION
**Pull request by stoprocent**
This PR replaces https://github.com/lake-wg/edhoc/pull/256 and is addressing this https://github.com/lake-wg/edhoc/issues/258

### CODE REVIEW

**Comment by stoprocent** [comment]
Code available here:
Ed25519 - https://github.com/stoprocent/edhoc/tree/feature/ed25519/test-vectors-11
P256 - https://github.com/stoprocent/edhoc/tree/feature/mbedtls/test-vectors-11

File: test-vectors-11/p256-vectors.txt [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+ID_CRED_I (CBOR Data Item) (3 bytes)
+a1 04 2b
+
+CRED_I (CBOR Data Item) (104 bytes)
+a2 02 77 34 32 2d 35 30 2d 33 31 2d 46 46 2d 45 46 2d 33 37 2d 33 32
+2d 33 39 08 a1 01 a4 01 01 02 2b 20 01 21 58 41 04 8a 93 ca 7e 1b c8
+46 47 d7 e7 eb 4c 61 07 c4 dc 4e 53 df 81 df d1 98 1c 7f 82 4a 7c 1b
+61 a6 fc 91 36 28 13 c2 5d b6 af 93 be 22 c3 50 ce b2 51 89 5b 9f 3a
+8d 85 a3 58 23 a2 22 2b 9d e2 c8 c8
+
```
Comment (marco-tiloca-sics) [inline test-vectors-11/p256-vectors.txt:244]: See the comment above about CRED_R. The same applies here to CRED_I.

File: test-vectors-11/p256-vectors.txt [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+ID_CRED_R (CBOR Data Item) (3 bytes)
+a1 04 32
+
+CRED_R (CBOR Data Item) (92 bytes)
+a2 02 6b 65 78 61 6d 70 6c 65 2e 65 64 75 08 a1 01 a4 01 01 02 32 20
+01 21 58 41 04 bb c3 49 60 52 6e a4 d3 2e 94 0c ad 2a 23 41 48 dd c2
+17 91 a1 2a fb cb ac 93 62 20 46 dd 44 f0 45 19 e2 57 23 6b 2a 0c e2
+02 3f 09 31 f1 f3 86 ca 7a fd a6 4f cd e0 10 8c 22 4c 51 ea bf 60 72
+
```
[the remark this answers is not in the document]
Comment (marco-tiloca-sics) [reply test-vectors-11/p256-vectors.txt:162]: Why the change to CRED_R (and CRED_I below) ? Now its content does not reflect the 'x' and 'y' coordinate of the public key. The value of CRED_R before this commit seemed correct.

**Comment by stoprocent** [comment]
I had to regenerate Initiator P256 Identity and I have updated G_R and G_I in vectors to only use x coordinate and have added a new field for G_R and G_I y coordinate.

**Comment by gselander** [comment]
@stoprocent @marco-tiloca-sics Should we have a short chat to sync the input with expectations :-)

**Comment by marco-tiloca-sics** [comment]
Just successfully validated the new trace for EdDSA with method 0 :-)

**Comment by stoprocent** [comment]
Yes, this was caused by just a typo which is now fixed. I was compressing the public key which was already compressed and some random bytes from memory were pulled.

**Comment by stoprocent** [comment]
@marco-tiloca-sics Nice!

**Comment by gselander** [comment]
Looks good, merging now.

**Comment by malishav** [comment]
thanks for providing these @stoprocent !

**Comment by marco-tiloca-sics** [comment]
Just checked the -traces draft and tested again. It looks good to me.

**Comment by gselander** [comment]
Thanks Marco!

**Pull request merged**